lib/msf/core/post/windows/dotnet.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+require 'msf/core/post/windows/registry'
+
+module Msf
+class Post
+module Windows
+
+#
+# This module serves to hold methods related to .NET framework
+#
+module Dotnet
+  include ::Msf::Post::Common
+  include ::Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super
+    register_advanced_options(
+      [
+        OptInt.new('Dotnet::Post::timeout', [true, 'Dotnet execution timeout, set < 0 to run async without termination', 15]),
+        OptBool.new('Dotnet::Post::log_output', [true, 'Write output to log file', false]),
+        OptBool.new('Dotnet::Post::dry_run', [true, 'Return encoded output to caller', false]),
+        OptBool.new('Dotnet::Post::force_wow64', [true, 'Force WOW64 execution', false])
+      ], 
+      self.class
+    )
+  end
+  #
+  # Searches the subkey for the value 'Version' which contains the
+  # actual version, rather than the over-arching release
+  # An alternative would be to query for it, and catch the exception.
+  #
+  
+  def search_for_version(dotnet_subkey)
+    dotnet_version = nil
+    begin
+      subkeys = registry_enumvals(dotnet_subkey)
+    rescue ::Exception => e
+      print_status("Encountered exception in search_for_version: #{e.class} #{e}")
+    end
+    subkeys.each do |i|
+      if i == 'Version'
+        dotnet_version = registry_getvaldata(dotnet_subkey, i)
+        break
+      end
+    end
+    return dotnet_version
+  end
+
+  #
+  # Bruteforce search all subkeys in an over-arching release to
+  # locate the actual release version.
+  #
+  def get_versionception(dotnet_vkey)
+    exact_version = nil
+    begin
+      subkeys = registry_enumkeys(dotnet_vkey)
+    rescue ::Exception => e
+      print_status("Encountered exception in get_versionception: #{e.class} #{e}")
+    end
+    subkeys.each do |i|
```

Comment:
Also need to check nil for ```subkeys``` here in case Rex::Post::Meterpreter::RequestError occurs.